src/payment-request.ts

```diff
@@ -27,54 +37,141 @@ export async function createPaymentRequest(
             payeeName: payee_name
             payeeEmail: payee_email
+            paymentAmount: payment_amount
             sessionPreviewData: session_preview_data
           }
         }
       `,
-      { sessionId, payeeName, payeeEmail, sessionPreviewData }
+      {
+        sessionId,
+        applicantName,
+        payeeName,
+        payeeEmail,
+        paymentAmount,
+        sessionPreviewData,
+      }
     );
   return response?.insert_payment_requests_one;
 }
 
-export function buildSessionPreviewData(
+export function extractSessionPreviewData(
   session: Session,
   sessionPreviewKeys: KeyPath[]
 ): PaymentRequest["sessionPreviewData"] {
   if (!session.data.passport?.data) {
     throw new Error("passport data not found");
   }
-  const data = session.data.passport.data!;
+  const passport = session.data.passport.data!;
   const sessionPreviewData: PaymentRequest["sessionPreviewData"] = {};
   sessionPreviewKeys.forEach((keyPath: KeyPath) => {
-    const value = keyPathAccessor(keyPath)(data);
+    const value = keyPathAccessor(keyPath)(passport);
     if (value === undefined) {
       const stringKey = keyPath.join(".");
-      throw new Error(`passport key ${stringKey} not found in passport data`);
+      throw new Error(`passport key "${stringKey}" not found in passport data`);
     }
     setByKeyPath(sessionPreviewData, keyPath, value as Value);
   });
   return sessionPreviewData;
 }
+
+// find the payment set in the passport
+async function getPaymentAmount(
+  client: GraphQLClient,
+  session: Session
+): Promise<number | undefined> {
+  if (!session.data?.breadcrumbs) {
+    throw new Error("breadcrumbs not found");
+  }
+  const breadcrumbs = session.data.breadcrumbs!;
+
+  if (!session.data?.passport?.data) {
+    throw new Error("passport not found");
+  }
+  const passport = session.data.passport.data!;
+
+  const flowId = session.flowId;
+  const flowGraph = await getLatestFlowGraph(client, flowId);
+  if (!flowGraph) {
+    throw new Error("flow not found");
+  }
+
+  const orderedBreadcrumbs = sortBreadcrumbs(flowGraph, breadcrumbs);
+  const lastCrumb = orderedBreadcrumbs.at(-1)!;
+
+  // if breadcrumbs includes the pay node
+  // use its key to find the pay amount in the passport
+  if (lastCrumb.type === ComponentType.Pay) {
+    const amountKey = getPaymentAmountKeyFromNode(
+      flowGraph[lastCrumb.id] as PayNode
+    );
+    return passport[amountKey];
+  }
+
+  // otherwise the pay node may not have been enterd into breadcrumbs so
+  // find the next node that is a pay component and use its key to find
+  // the pay amount in the passport
+  for (const node of sortFlow(flowGraph)) {
+    const isNextNodeAfterLastCrumb = node.parentId === lastCrumb.id;
+    if (isNextNodeAfterLastCrumb && node.type === ComponentType.Pay) {
+      const amountKey = getPaymentAmountKeyFromNode(
+        flowGraph[node.id] as PayNode
+      );
+      return passport[amountKey];
+    }
+  }
```

Comment:
This function looks up the payment amount key from either the last breadcrumb or the next breadcrumb that is also a Pay component. I could also update the breadcrumbs here but decided against it in this PR.